File: core/src/main/java/org/elkd/core/consensus/RaftFactory.java
package org.elkd.core.consensus;

import org.elkd.core.config.Config;
import org.elkd.core.consensus.messages.Entry;
import org.elkd.core.log.LogProvider;
import org.elkd.core.server.cluster.ClusterMessenger;

public class RaftFactory {
  private RaftFactory() { }

  public static Raft create(final Config config,
                            final LogProvider<Entry> logProvider,
                            final ClusterMessenger clusterMessenger) {
    return new Raft(
        config,
        clusterMessenger,
        new RaftContext(),
        logProvider
    );
  }
}

File: core/src/main/java/org/elkd/core/consensus/RaftFactory.kt
package org.elkd.core.consensus

import org.elkd.core.config.Config
import org.elkd.core.runtime.TopicModule
import org.elkd.core.server.cluster.ClusterMessenger

object RaftFactory {
  fun create(config: Config,
             topicModule: TopicModule,
             clusterMessenger: ClusterMessenger): Raft {
    return Raft(
        config,
        clusterMessenger,
        RaftContext(),
        topicModule
    )
  }
}
